### ISSUE TITLE
running upgrade-requirements give an TypeError

### ISSUE THREAD

**Issue by bakkerthehacker** [Yelp/requirements-tools#15]
Running `upgrade-requirements` gives the following error:

```
*** exec-ing: to use the virtualenv python
>>> /nail/tmp/tmpfbx1r71c/venv/bin/python /nail/home/gbakker/pg/services/payload/virtualenv_run/lib/python3.5/site-packages/requirements_tools/upgrade_requirements.py --tempdir /nail/tmp/tmpfbx1r71c --index-url https://pypi.yelpcorp.com/simple/ --exec-count 0 --exec-limit 10
Traceback (most recent call last):
  File "/nail/home/gbakker/pg/services/payload/virtualenv_run/lib/python3.5/site-packages/requirements_tools/upgrade_requirements.py", line 185, in <module>
    exit(main())
  File "/nail/home/gbakker/pg/services/payload/virtualenv_run/lib/python3.5/site-packages/requirements_tools/upgrade_requirements.py", line 148, in main
    reqs = installed('requirements-minimal.txt')
  File "/nail/home/gbakker/pg/services/payload/virtualenv_run/lib/python3.5/site-packages/requirements_tools/upgrade_requirements.py", line 69, in installed
    unmet.add('{}{}'.format(sub.key, ','.join(sub.specifier)))
TypeError: sequence item 0: expected str instance, Specifier found
```

**Comment by asottile** [comment]
Ah, I'm able to reproduce with the following packages:

```
==> pkg2/setup.py <==
from setuptools import setup

setup(
    name='pkg2',
    version='3',
    install_requires=['pkg3[myextra]'],
)

==> pkg3/setup.py <==
from setuptools import setup

setup(
    name='pkg3',
    version='3',
    extras_require={
        'myextra': ['simplejson>0'],
    },
)

==> pkg/setup.py <==
from setuptools import setup

setup(
    name='pkg1',
    version='4',
    install_requires=['pkg3'],
)
```

And 

```
$ tail -n999 requirements*minimal.txt
==> requirements-dev-minimal.txt <==

==> requirements-minimal.txt <==
pkg1
```

Should be an easy fix, harder part will be incorporating this as a regression test :)

**Comment by asottile** [comment]
actually lol, upgrade-requirements never got a testsuite, I'll just fix the bug

**Comment by asottile** [comment]
This fix is now available in 1.1.2 (you'll want to add that to yelp's internal pypi probably through `fetch-python-package requirements-tools` (if @chriskuehl  hasn't already))
